File: go/pe019.go
package main

import (
	"fmt"
)

const ROWS = 15

func main() {
	cycle := []int{31, 28, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31,
		31, 28, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31,
		31, 28, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31,
		31, 29, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31}

	var count int
	dow := 2 // start on tuesday 1901.1.1
	// 2000 is a leap year too!
	for i := 0; i < 25; i++ {
		for j := 0; j < len(cycle); j++ {
			if dow == 0 {
				count += 1
			}
			dow = (dow + cycle[j]) % 7
		}
	}
	fmt.Printf("%d\n", count)
}

File: ruby/pe019.rb
#!/usr/bin/ruby

cycle = Array[31, 28, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31,
           31, 28, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31,
           31, 28, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31,
           31, 29, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31]

count = 0
dow = 2  # start on tuesday 1901.1.1
         # 2000 is a leap year too!
(0 ... 25).each do |i|
  cycle.each do |j|
    if dow == 0
      count += 1
    end
    dow = (dow + j) % 7
  end
end

puts(count)
